BUILD ERROR: diagnostic_msgs headers not properly exported

When building custom mavros_plugins you have to include the `mavros_uas.hpp` header. But when I do I get this error:

```
In file included from mavros/include/mavros/mavros_uas.hpp:43,

/opt/ros/jazzy/include/diagnostic_updater/diagnostic_updater.hpp:46:10: fatal error: diagnostic_msgs/msg/diagnostic_array.hpp: No such file or directory
   46 | #include "diagnostic_msgs/msg/diagnostic_array.hpp"
```

I can work around this by adding a dependency on `diagnostic_msgs` in my custom package, but obviously this shouldn't be necessary. I'll see if I can fix the upstream CMakeList.txt, but in case I don't get around to it I'd like to at least make this issue.

--- a/mavros/CMakeLists.txt
+++ b/mavros/CMakeLists.txt
@@ -121,6 +121,7 @@ add_library(mavros SHARED
 target_link_libraries(mavros PUBLIC
   ${mavros_msgs_TARGETS}
   ${sensor_msgs_TARGETS}
+  ${diagnostic_msgs_TARGETS}
   diagnostic_updater::diagnostic_updater
   pluginlib::pluginlib
   rclcpp::rclcpp
@@ -181,6 +182,7 @@ target_link_libraries(mavros_plugins PUBLIC
   ${std_msgs_TARGETS}
   ${std_srvs_TARGETS}
   ${trajectory_msgs_TARGETS}
+  ${diagnostic_msgs_TARGETS}
   angles::angles
   diagnostic_updater::diagnostic_updater
   geographic_msgs::validation

--- a/mavros/launch/test_compose.launch.py
+++ b/mavros/launch/test_compose.launch.py
@@ -5,7 +5,6 @@ from launch_ros.descriptions import ComposableNode
 
 def generate_launch_description():
     """Generate launch description for MAVROS composable node."""
-
     container = ComposableNodeContainer(
         name="mavros_container",
         namespace="",

--- a/mavros_extras/CMakeLists.txt
+++ b/mavros_extras/CMakeLists.txt
@@ -150,6 +150,7 @@ target_link_libraries(mavros_extras_plugins PUBLIC
   ${std_msgs_TARGETS}
   ${std_srvs_TARGETS}
   ${trajectory_msgs_TARGETS}
+  ${diagnostic_msgs_TARGETS}
   angles::angles
   diagnostic_updater::diagnostic_updater
   geographic_msgs::validation

